storage/replica.go

```diff
@@ -191,16 +192,20 @@ func NewReplica(desc *roachpb.RangeDescriptor, store *Store) (*Replica, error) {
 		tsCache:     NewTimestampCache(store.Clock()),
 		sequence:    NewSequenceCache(desc.RangeID),
 		pendingCmds: map[cmdIDKey]*pendingCmd{},
+		RangeID:     desc.RangeID,
+	}
+	r.Lock()
```

Comment:
There are a lot of early returns between this lock and its unlock. The unlock needs to be deferred, or we can (I think) rely on the fact that in the constructor no other thread can have a reference to this object yet and initialize it without locking.